From stable-bounces@linux.kernel.org Mon Sep 25 22:30:41 2006
Date: Mon, 25 Sep 2006 22:29:54 -0700 (PDT)
Message-Id: <20060925.222954.59655772.davem@davemloft.net>
To: stable@kernel.org
From: Fabio Olive Leite <fleite@redhat.com>
Subject: IPV6: bh_lock_sock_nested on tcp_v6_rcv

From: Fabio Olive Leite <fleite@redhat.com>

A while ago Ingo patched tcp_v4_rcv on net/ipv4/tcp_ipv4.c to use
bh_lock_sock_nested and silence a lock validator warning. This fixed
it for IPv4, but recently I saw a report of the same warning on IPv6.

Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/ipv6/tcp_ipv6.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.18.orig/net/ipv6/tcp_ipv6.c
+++ linux-2.6.18/net/ipv6/tcp_ipv6.c
@@ -1228,7 +1228,7 @@ process:
 
 	skb->dev = NULL;
 
-	bh_lock_sock(sk);
+	bh_lock_sock_nested(sk);
 	ret = 0;
 	if (!sock_owned_by_user(sk)) {
 #ifdef CONFIG_NET_DMA
